### PR TITLE
Fix daily build with nightly

### PR DESCRIPTION
**Pull request by yogh333**

### CODE REVIEW

File: .github/workflows/build_with_current_nightly.yml [hunk 1]
```diff
@@ -18,9 +18,9 @@ jobs:
       - name: update nightly toolchain
         run: |
           rustup update nightly
-          rustup default nightly
           rustup component add rust-src --toolchain nightly
-          cargo ledger setup
+          export RUST_NIGHTLY=nightly
```
Comment (xchapron-ledger) [inline .github/workflows/build_with_current_nightly.yml:22]: Isn't this necessary at build step too?
I'm afraid it's not persisted between steps?

File: .github/workflows/build_with_current_nightly.yml [hunk 2]
```diff
@@ -33,6 +32,7 @@ jobs:
           BUILD_DEVICE_NAME="$(echo ${{ matrix.device }})"
           BIN_DIR_NAME="$(echo ${{ matrix.device }} | sed 's/nanosplus/nanos2/')"
           cd app-boilerplate
+          RUST_NIGHTLY=nightly
           cargo ledger build ${{ matrix.device }}
```
Comment (xchapron-ledger) [inline .github/workflows/build_with_current_nightly.yml:36]: Would `cargo +nightly ledger build` work here?

Comment (yogh333) [reply .github/workflows/build_with_current_nightly.yml:36]: Nope, the nightly version set by RUST_NIGHTLY would be used in that case.

**Review by xchapron-ledger — APPROVED** [review]

File: .github/workflows/build_with_current_nightly.yml (hunk 2 above)
Comment (yogh333) [reply .github/workflows/build_with_current_nightly.yml:36]:
```
cargo +nightly ledger build nanosplus
Use Rust nightly toolchain: nightly-2023-11-10
Updating crates.io index
    Updating git repository `https://github.com/rust-embedded-community/serde-json-core`
  Downloaded autocfg v1.1.0
  Downloaded shlex v1.2.0
  Downloaded quote v1.0.33
  Downloaded peeking_take_while v0.1.2
  Downloaded color_quant v1.1.0
  Downloaded lazy_static v1.4.0
  Downloaded stable_deref_trait v1.2.0
  Downloaded ledger_secure_sdk_sys v1.2.0
...
```

**Pull request merged**